internal/api/github/client.go

```diff
@@ -58,14 +58,13 @@ func (c *Client) GetConfiguration(ctx context.Context, ownerLogin, repoName stri
 		ownerLogin,
 		repoName,
 		configuration.ConfigurationFilePath,
-		&github.RepositoryContentGetOptions{Ref: defaultBranch},
+		nil,
 	)
 
-	if resp.StatusCode == http.StatusNotFound {
-		return nil, ErrNoConfigurationFile
-	}
-
 	if err != nil {
+		if resp != nil && resp.StatusCode == http.StatusNotFound {
```

Comment:
I don't get why this _if_ condition is nested? shouldn't it be moved beneath?

```go
if err != nil {
  return nil, fmt.Errorf("error downloading configuration: %w", err)
}

if resp != nil && resp.StatusCode == http.StatusNotFound {
  return nil, ErrNoConfigurationFile
}
```

Reply:
no, it should be like that as when "Not Found" is reported both error and status code are reported, but GTA wants to detect this particular error case and report status 204, in your scenario we would always fall into first condition and never report `ErrNoConfigurationFile`